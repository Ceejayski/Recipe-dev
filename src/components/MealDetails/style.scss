.card-title {
  font-size: 0.9em !important;
  font-weight: bolder;
}

.card-title,
.list-title {
  font-family: "Merienda", cursive;
}

.list-title {
  font-size: 0.7em !important;
}

.meal-details {
  transition: all 0.2s ease-in-out;
}

.meal-details:hover {
  transform: scale(1.1);
}
