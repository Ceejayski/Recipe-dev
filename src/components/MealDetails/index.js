import React from 'react';
import PropTypes from 'prop-types';
import './style.scss';
import { Card, ListGroup } from 'react-bootstrap';
import { Link } from 'react-router-dom';

function MealDetails({ data, tags }) {
  return (
    <div className="meal-details">
      <Link to={`/recipe/${data.idMeal}`}>
        <Card bg="white" className="rounded">
          <Card.Img variant="top" src={data.strMealThumb} />
          <Card.Body>
            <Card.Title>{data.strMeal}</Card.Title>
          </Card.Body>
          {tags && (
          <ListGroup variant="flush">
            {data.strCategory !== undefined && (
            <ListGroup.Item>
              <div className="details d-flex justify-content-between">
                <p className="mb-0 list-title">Main Ingredient</p>
                <p className="mb-0" style={{ maxWidth: '60%' }}>{data.strCategory}</p>
              </div>
            </ListGroup.Item>
            )}
            {(data.strTags !== null) && (
            <ListGroup.Item>
              <div className="details d-flex justify-content-between">
                <p className="mb-0 list-title">Tags</p>
                <p className="mb-0" style={{ maxWidth: '70%' }}>{data.strTags}</p>
              </div>
            </ListGroup.Item>
            )}
            {data.strArea !== undefined && (
            <ListGroup.Item>
              <div className="details d-flex justify-content-between">
                <p className="mb-0 list-title">Localization</p>
                <p className="mb-0" style={{ maxWidth: '70%' }}>{data.strArea}</p>
              </div>
            </ListGroup.Item>
            )}
          </ListGroup>
          )}
        </Card>
      </Link>
    </div>
  );
}

MealDetails.defaultProps = {
  tags: true,
};

MealDetails.propTypes = {
  data: PropTypes.shape().isRequired,
  tags: PropTypes.bool,
};

export default MealDetails;
