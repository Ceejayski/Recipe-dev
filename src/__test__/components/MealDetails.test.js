import { BrowserRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';
import MealDetails from '../../components/MealDetails';

test('renders Correctly', () => {
  const data = { idMeal: '1', strMealThumb: 'img', strMeal: 'name' };
  const Placeholder = () => (
    <BrowserRouter>
      <MealDetails data={data} />
    </BrowserRouter>
  );
  const tree = create(<Placeholder />).toJSON();
  expect(tree).toMatchSnapshot();
});
