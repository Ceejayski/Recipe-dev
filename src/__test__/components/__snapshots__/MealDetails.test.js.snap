// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Correctly 1`] = `
<div
  className="meal-details"
>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    <div
      className="rounded card bg-white"
    >
      <img
        className="card-img-top"
        src="img"
      />
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          name
        </div>
      </div>
      <div
        className="list-group list-group-flush"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          <div
            className="details d-flex justify-content-between"
          >
            <p
              className="mb-0 list-title"
            >
              Tags
            </p>
            <p
              className="mb-0"
              style={
                Object {
                  "maxWidth": "70%",
                }
              }
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
